lib/Bridge/TolerantParser/WorseReflection/DoctrineAnnotationCompletor.php

```diff
@@ -0,0 +1,139 @@
+    }
+
+    public function complete(TextDocument $source, ByteOffset $byteOffset): Generator
+    {
+        $truncatedSource = $this->truncateSource((string) $source, $byteOffset->toInt());
+        $sourceNodeFile = $this->parser->parseSourceFile((string) $source);
+
+        $node = $this->findNodeForPhpdocAtPosition(
+            $sourceNodeFile,
+            // the parser requires the byte offset, not the char offset
+            strlen($truncatedSource)
+        );
+
+        if (!$node) {
+            // Ignore this case is the cursor is not in a phpdoc block
+            return true;
+        }
+
+        if (!$annotation = $this->extractAnnotation($truncatedSource)) {
+            // Ignore if not an annotation
+            return true;
+        }
+
+        $suggestions = $this->completeName($annotation);
+
+        foreach ($suggestions as $suggestion) {
+            if (!$this->isAnAnnotation($suggestion)) {
+                continue;
+            }
+
+            yield $suggestion;
+        }
+
+        return $suggestions->getReturn();
+    }
+
+    protected function createSuggestionOptions(NameSearchResult $result): array
+    {
+        return array_merge(parent::createSuggestionOptions($result), [
+            'snippet' => (string) $result->name()->head() .'($1)$0',
+        ]);
+    }
+
+    private function truncateSource(string $source, int $byteOffset): string
+    {
+        // truncate source at byte offset - we don't want the rest of the source
+        // file contaminating the completion (for example `$foo($<>\n    $bar =
+        // ` will evaluate the Variable node as an expression node with a
+        // double variable `$\n    $bar = `
+        $truncatedSource = substr($source, 0, $byteOffset);
+
+        // determine the last non-whitespace _character_ offset
+        $characterOffset = OffsetHelper::lastNonWhitespaceCharacterOffset($truncatedSource);
+
+        // truncate the source at the character offset
+        $truncatedSource = mb_substr($source, 0, $characterOffset);
+
+        return $truncatedSource;
+    }
+
+    private function findNodeForPhpdocAtPosition(SourceFileNode $sourceNodeFile, int $position): ?Node
+    {
+        /** @var Node $node */
+        foreach ($sourceNodeFile->getDescendantNodes() as $node) {
+            if (
+                $node->getFullStart() < $position
+                && $position < $node->getStart()
+            ) {
+                // If the text is a phpdoc block return the node
+                return $node->getDocCommentText() ? $node : null;
+            }
+        }
+
+        return null;
+    }
+
+    private function isAnAnnotation(Suggestion $suggestion): bool
+    {
+        try {
+            $reflectionClass = $this->reflector->reflectClass($suggestion->classImport());
+            $docblock = $reflectionClass->docblock();
+
+            return false !== strpos($docblock->raw(), '@Annotation');
+        } catch (\Throwable $error) {
```

Comment:
Use `Phpactor\WorseReflection\Core\Exception\NotFound` exception if that's what we're expecting (no other exception should be expected)

Reply:
I don't expect any error, I was over protective.
So I only catch `NotFound` and if there is an other error (which should not) then it will be catch later on.